core/kazoo_proper/src/pqc_cb_phone_numbers.erl

```diff
@@ -0,0 +1,298 @@
+    pqc_util:transition_if(Model
+                          ,[{fun pqc_kazoo_model:does_account_exist/2, [AccountId]}
+                           ,{fun pqc_kazoo_model:is_number_in_account/3, [AccountId, Number]}
+                           ,{fun pqc_kazoo_model:transition_number_state/3, [Number, APIResp]}
+                           ]
+                          );
+next_state(Model
+          ,_APIResp
+          ,{'call', _, 'list_number', [_API, _AccountId, _Number]}
+          ) ->
+    Model.
+
+-spec postcondition(pqc_kazoo_model:model(), any(), any()) -> boolean().
+postcondition(Model
+             ,{'call', _, 'create_account', [_API, Name]}
+             ,APIResult
+             ) ->
+    case pqc_kazoo_model:account_id_by_name(Model, Name) of
+        'undefined' ->
+            'undefined' =/= pqc_cb_response:account_id(APIResult);
+        _AccountId ->
+            500 =:= pqc_cb_response:error_code(APIResult)
+    end;
+postcondition(Model
+             ,{'call', _, 'list_number', [_API, AccountId, Number]}
+             ,APIResult
+             ) ->
+    case pqc_kazoo_model:does_account_exist(Model, AccountId) of
+        'false' -> ?FAILED_RESPONSE =:= APIResult;
+        'true' ->
+            case pqc_kazoo_model:number_data(Model, Number) of
+                'undefined' ->
+                    404 =:= pqc_cb_response:error_code(APIResult);
+                #{'number_state' := NumberState
+                 ,'account_id' := AccountId
+                 } ->
+                    NumberState =:= pqc_cb_response:number_state(APIResult);
+                _N ->
+                    404 =:= pqc_cb_response:error_code(APIResult)
+            end
+    end;
+postcondition(Model
+             ,{'call', _, 'add_number', [_API, AccountId, Number]}
+             ,APIResult
+             ) ->
+    case pqc_kazoo_model:does_account_exist(Model, AccountId) of
+        'false' -> ?FAILED_RESPONSE =:= APIResult;
+        'true' ->
+            case pqc_kazoo_model:number_data(Model, Number) of
+                #{'account_id' := AccountId
+                 ,'number_state' := _CurrentState
+                 } ->
+                    409 =:= pqc_cb_response:error_code(APIResult)
+                        andalso <<"number_exists">> =:= pqc_cb_response:message(APIResult);
+                #{} ->
+                    %% if in another account
+                    409 =:= pqc_cb_response:error_code(APIResult);
+                'undefined' ->
+                    case <<"success">> =:= pqc_cb_response:status(APIResult) of
+                        'true' -> 'reserved' =:= pqc_cb_response:number_state(APIResult);
+                        'false' -> 500 =:= pqc_cb_response:error_code(APIResult)
+                    end
+            end
+    end;
+postcondition(Model
+             ,{'call', _, 'remove_number', [_API, AccountId, Number]}
+             ,APIResult
+             ) ->
+    case pqc_kazoo_model:does_account_exist(Model, AccountId) of
```

Comment:
Agreed. I tried to keep it simple to make sure failing runs could be debugged easily while I worked out the kinks (hence one account, one number for the time being). Not married to how predicates are defined for post-conditions or next_state transitions; will let them evolve as this moves forward.